### PR TITLE
Implement seekable

### PR DESCRIPTION
**Pull request by dmlap**
Override the Flash tech's seekable method to take into account live playlists.

### CODE REVIEW

**Comment by dmlap** [comment]
This is dependent on https://github.com/videojs/video.js/commit/1c35bfafc2d6d65f10f86c0f5918e07e3fd4c4e0

File: src/playlist.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+   * for seeking
+   */
+  seekable = function(playlist) {
+    var startOffset, targetDuration;
+    // without segments, there are no seekable ranges
+    if (!playlist.segments) {
+      return videojs.createTimeRange();
+    }
+    // when the playlist is complete, the entire duration is seekable
+    if (playlist.endList) {
```
Comment (gkatsev) [inline src/playlist.js]: could it be that you get an `endList` in a sliding window manifest? In which case, the start range shouldn't be zero.

Comment (gkatsev) [inline src/playlist.js]: Or would this only come up in the DVR/VOD case only?

**Comment by gkatsev** [comment]
Otherwise, LGTM

File: src/playlist.js (hunk 1 above)
Comment (dmlap) [inline src/playlist.js]: `endList` is the signal that the playlist is complete and there will be no further updates, i.e. it's now a VOD playlist.

Comment (gkatsev) [inline src/playlist.js]: How do we know that a sliding window has ended, assuming it isn't turning into a VOD playlist?

**Comment by forbesjo** [comment]
LGTM

**Pull request merged**